.ci/bumpStackVersion.groovy

```diff
@@ -99,50 +99,50 @@ def bumpStackVersion(Map args = [:]){
   def scriptFile = args.containsKey('scriptFile') ? args.get('scriptFile') : error('bumpStackVersion: scriptFile argument is required')
   def branch = args.containsKey('branch') ? args.get('branch') : error('bumpStackVersion: branch argument is required')
   def reusePullRequest = args.get('reusePullRequest', false)
-  def labels = args.get('labels', '')
+  def labels = args.get('labels', '').replaceAll('\\s','')
   log(level: 'INFO', text: "bumpStackVersion(repo: ${repo}, branch: ${branch}, scriptFile: ${scriptFile}, reusePullRequest: ${reusePullRequest}, labels: '${labels}')")
 
   def branchName = findBranch(branch: branch, versions: latestVersions)
   def versionEntry = latestVersions.get(branchName)
   def message = createPRDescription(versionEntry)
- 
-  catchError(buildResult: 'SUCCESS', stageResult: 'UNSTABLE') {
-    deleteDir()
-    setupAPMGitEmail(global: true)
-    git(url: "https://github.com/elastic/${repo}.git", branch: branchName, credentialsId: '2a9602aa-ab9f-4e52-baf3-b71ca88469c7-UserAndToken')
-    sh(script: "${scriptFile} '${versionEntry.build_id}'", label: "Prepare changes for ${repo}")
-
-    pullRequest(reusePullRequest: reusePullRequest,
-                stackVersion: versionEntry.build_id,
-                message: message,
-                labels: labels)
-  }
-}
-
-def pullRequest(Map args = [:]){
-  def stackVersion = args.stackVersion
-  def message = args.message
-  def labels = args.labels.replaceAll('\\s','')
-  def reusePullRequest = args.get('reusePullRequest', false)
+  def stackVersion = versionEntry.build_id
+  def title = "bump: stack version"
   if (labels.trim()) {
-    labels = "automation,${labels}"
+    labels = "automation,dependency,${labels}"
   }
 
-  if (params.DRY_RUN_MODE) {
-    log(level: 'INFO', text: "DRY-RUN: pullRequest(stackVersion: ${stackVersion}, reusePullRequest: ${reusePullRequest}, labels: ${labels}, message: '${message}')")
-    return
+  catchError(buildResult: 'SUCCESS', stageResult: 'UNSTABLE') {
+    prepareContext(repo: repo, branchName: branchName)
+    if (reusePullRequest && reusePullRequestIfPossible(title: title, labels: labels)) {
+      try {
+        sh(script: "${scriptFile} '${versionEntry.build_id}' 'noBranchCreation'", label: "Prepare changes for ${repo}")
```

Comment:
In order to support the scenario of reusing ane xisting PR it will require to support a new flag in the scripts.

The one in charge of pushing in a new PR (therefore it requires a new branch) or reuse the existing one, which therefore won't require a branch if using `gh pr checkout <number>`